Sources/XcodeProj/Workspace/XCWorkspaceData.swift

```diff
@@ -28,13 +28,17 @@ extension XCWorkspaceData: Writable {
             throw XCWorkspaceDataError.notFound(path: path)
         }
 
-        let xml = try AEXMLDocument(xml: path.read())
-        let children = try xml
-            .root
-            .children
-            .compactMap(XCWorkspaceDataElement.init(element:))
-
-        self.init(children: children)
+        do {
+            let xml = try AEXMLDocument(xml: path.read())
+            let children = try xml
+                .root
+                .children
+                .compactMap(XCWorkspaceDataElement.init(element:))
+
+            self.init(children: children)
+        } catch AEXMLError.parsingFailed {
+            fatalError("We could not parse the file \(path.lastComponent). Make sure it's a valid XML/Plist/JSON file and try again")
```

Comment:
I'd refrain from `fatalError` here because the caller might want to catch the error and recover from it. Therefore, I'd suggest to define a `XCWorkspaceDataError.parseError` error type that you can throw from here. That approach also makes testing easier.